### PR TITLE
PROGRAMMES-5824 tests ProgrammesService

### PR DESCRIPTION
**Pull request by tatitati**

### CODE REVIEW

File: tests/Service/ProgrammesService/CountAllTleosByCategoryTest.php [hunk 1]
```diff
@@ -2,21 +2,27 @@
 
 namespace Tests\BBC\ProgrammesPagesService\Service\ProgrammesService;
 
+use BBC\ProgrammesPagesService\Domain\Entity\Genre;
+
 class CountAllTleosByCategoryTest extends AbstractProgrammesServiceTest
 {
-    public function testCountAllTleosByCategory()
+    public function testProtocolWithRepository()
```
Comment (BjornTwachtmann) [inline tests/Service/ProgrammesService/CountAllTleosByCategoryTest.php]: Not quite sure what this test name means, can you make it clearer?

Comment (tatitati) [reply tests/Service/ProgrammesService/CountAllTleosByCategoryTest.php]: Yes of course! jeje

File: tests/Service/ProgrammesService/CountEpisodeGuideChildrenTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $programme = $this->createMock(Programme::class);
+
+        $this->mockRepository
+            ->method('countEpisodeGuideChildren')
+            ->will($this->onConsecutiveCalls(0, 1, 10));
+
+        $this->assertEquals(0, $this->service()->countEpisodeGuideChildren($programme));
+        $this->assertEquals(1, $this->service()->countEpisodeGuideChildren($programme));
+        $this->assertEquals(10, $this->service()->countEpisodeGuideChildren($programme));
```
Comment (BjornTwachtmann) [inline tests/Service/ProgrammesService/CountEpisodeGuideChildrenTest.php]: I think this style is a little unclear, a dataprovider might be better than having multiple tests in the same method

Comment (tatitati) [reply tests/Service/ProgrammesService/CountEpisodeGuideChildrenTest.php]: yeah, I was expecting a possible comment about it. OK, I will, I aggree.

**Comment by BjornTwachtmann** [comment]
Other than those two comments, looks good

**Pull request merged**